coalib/results/result_actions/IgnoreResultAction.py

```diff
@@ -16,11 +16,18 @@ class IgnoreResultAction(ResultAction):
 
     @staticmethod
     @enforce_signature
-    def is_applicable(result: Result, original_file_dict, file_diff_dict):
+    def is_applicable(result: Result,
+                      original_file_dict,
+                      file_diff_dict,
+                      applied_actions=()):
         """
         For being applicable, the result has to point to a number of files
         that have to exist i.e. have not been previously deleted.
+        Additionally, the action should not have been applied to the current
+        result before.
         """
+        if any(isinstance(x, IgnoreResultAction) for x in applied_actions):
+            return False
```

Comment:
docs say you should only return True or a string, this is breaking that contract. I.e. there should always be a string message with a reason, also in this case